### PR TITLE
C#: Remove legacy `odasa` support

### PR DESCRIPTION
**Pull request by hvitved**
The following environment variables are no longer supported:

```
ODASA_BUILD_ERROR_DIR
ODASA_CSHARP_LAYOUT
ODASA_SNAPSHOT
SEMMLE_DIST
SEMMLE_EXTRACTOR_OPTIONS
SEMMLE_PLATFORM_TOOLS
SEMMLE_PRESERVE_SYMLINKS
SOURCE_ARCHIVE
TRAP_FOLDER
```

### CODE REVIEW

File: csharp/autobuilder/Semmle.Autobuild.Shared/Autobuilder.cs [hunk 1]
```diff
@@ -190,18 +190,14 @@ protected Autobuilder(IBuildActions actions, AutobuildOptions options)
...
             TrapDir =
                 Actions.GetEnvironmentVariable($"CODEQL_EXTRACTOR_{this.Options.Language.UpperCaseName}_TRAP_DIR") ??
-                Actions.GetEnvironmentVariable("TRAP_FOLDER") ??
-                throw new InvalidEnvironmentException($"The environment variable CODEQL_EXTRACTOR_{this.Options.Language.UpperCaseName}_TRAP_DIR or TRAP_FOLDER has not been set.");
+                throw new InvalidEnvironmentException($"The environment variable CODEQL_EXTRACTOR_{this.Options.Language.UpperCaseName}_TRAP_DIR has not been set.");
 
             SourceArchiveDir =
                 Actions.GetEnvironmentVariable($"CODEQL_EXTRACTOR_{this.Options.Language.UpperCaseName}_SOURCE_ARCHIVE_DIR") ??
-                Actions.GetEnvironmentVariable("SOURCE_ARCHIVE") ??
                 throw new InvalidEnvironmentException($"The environment variable CODEQL_EXTRACTOR_{this.Options.Language.UpperCaseName}_SOURCE_ARCHIVE_DIR or SOURCE_ARCHIVE has not been set.");
```
Comment (michaelnebel) [inline csharp/autobuilder/Semmle.Autobuild.Shared/Autobuilder.cs:201]: `SOURCE_ARCHIVE` needs to be removed from exception message.

File: csharp/autobuilder/Semmle.Autobuild.Shared/BuildScript.cs [hunk 2]
```diff
@@ -70,7 +70,7 @@ public BuildCommand(string exe, string? argumentsOpt, bool silent, string? worki
                 this.environment = environment;
             }
 
-            public override string ToString() => exe + " " + arguments;
+            public override string ToString() => arguments.Length > 0 ? exe + " " + arguments : exe;
```
Comment (michaelnebel) [inline csharp/autobuilder/Semmle.Autobuild.Shared/BuildScript.cs:73]: Maybe use !string.IsNullOrEmpty.

**Review by michaelnebel — APPROVED** [review]
LGTM. Very nice with some cleanup.
Have added a couple of comments. Nothing that should block a merge.

File: csharp/autobuilder/Semmle.Autobuild.Shared/Autobuilder.cs (hunk 1 above)
Comment (hvitved) [reply csharp/autobuilder/Semmle.Autobuild.Shared/Autobuilder.cs:201]: Good spot.

File: csharp/autobuilder/Semmle.Autobuild.Shared/BuildScript.cs (hunk 2 above)
Comment (hvitved) [reply csharp/autobuilder/Semmle.Autobuild.Shared/BuildScript.cs:73]: `arguments` cannot be `null` (we are using C# 8 nullable reference types), so we only need the emptiness check.

Comment (michaelnebel) [reply csharp/autobuilder/Semmle.Autobuild.Shared/BuildScript.cs:73]: Yes, let's keep it as it is :-)

**Review by michaelnebel — APPROVED** [review]
Very nice!

**Pull request merged**